gradio/components/custom_html_components/audio_gallery.py

```diff
@@ -0,0 +1,170 @@
+                audioItems.forEach(i => i.removeAttribute('data-selected'));
+                item.setAttribute('data-selected', 'true');
+                props.value = audio.src;
+            });
+            
+            playBtn.addEventListener('click', (e) => {
+                e.stopPropagation();
+                if (audio.paused) {
+                    document.querySelectorAll('.audio-item audio').forEach(a => a.pause());
+                    document.querySelectorAll('.play-btn').forEach(b => b.textContent = '▶');
+                    audio.play();
+                    playBtn.textContent = '⏸';
+                } else {
+                    audio.pause();
+                    playBtn.textContent = '▶';
+                }
+            });
+            
+            audio.addEventListener('timeupdate', () => {
+                const currentTime = Math.floor(audio.currentTime);
+                const minutes = Math.floor(currentTime / 60);
+                const seconds = currentTime % 60;
+                timeDisplay.textContent = `${minutes}:${seconds.toString().padStart(2, '0')}`;
+                
+                const progress = audio.currentTime / audio.duration;
+                drawWaveform(canvas, ctx, progress);
+            });
+            
+            audio.addEventListener('ended', () => {
+                playBtn.textContent = '▶';
+                drawWaveform(canvas, ctx, 0);
+            });
+        });
+        
+        function drawWaveform(canvas, ctx, progress = 0) {
+            const width = canvas.width;
+            const height = canvas.height;
+            const bars = 50;
+            const barWidth = width / bars;
+            
+            ctx.clearRect(0, 0, width, height);
+            
+            for (let i = 0; i < bars; i++) {
+                const barHeight = (Math.sin(i * 0.5) * 0.3 + Math.random() * 0.7) * height * 0.8;
+                const x = i * barWidth;
+                const y = (height - barHeight) / 2;
+                
+                ctx.fillStyle = i / bars < progress ? '#FF7C00' : '#ccc';
+                ctx.fillRect(x, y, barWidth - 2, barHeight);
+            }
+        }
+        """
+
+        super().__init__(
+            value=value or (audio_urls[0] if audio_urls else None),
+            html_template=html_template,
+            css_template=css_template,
+            js_on_load=js_on_load,
+            audio_urls=audio_urls,
+            labels=labels,
+            columns=columns,
+            label=label,
+            **kwargs,
+        )
+
+    def api_info(self):
+        return {
+            "type": "string",
+            "title": "Audio URL",
+        }
+
+
+if __name__ == "__main__":
+    with gr.Blocks() as demo:
+        gr.Markdown("# Audio Gallery Demo")
+
+        gallery = AudioGallery(
+            audio_urls=[
+                "https://github.com/gradio-app/gradio/raw/main/test/test_files/audio_sample.wav",
```

Comment:
Feel like this would be a better demo with just 2 different files rather than 6 identical files